CMake install target breaks build on CMake 3.7

Maybe wrong name?
```
CMake Error at CMakeLists.txt:32 (install):
  install TARGETS given target "IrrlichtMt" which does not exist in this directory.
-- Configuring incomplete, errors occurred!
```

```
$ cmake .. -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=/opt/minetest                                                   
-- The CXX compiler identification is GNU 6.3.0                                 
-- Check for working CXX compiler: /usr/bin/c++                                 
-- Check for working CXX compiler: /usr/bin/c++ -- works                        
-- Detecting CXX compiler ABI info      
-- Detecting CXX compiler ABI info - done                                       
-- Detecting CXX compile features       
-- Detecting CXX compile features - done                                        
-- *** Building IrrlichtMt 1.9.0 ***    
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.8")        
-- Found JPEG: /usr/lib/x86_64-linux-gnu/libjpeg.so                             
-- Found PNG: /usr/lib/x86_64-linux-gnu/libpng.so (found version "1.6.28")      
-- Looking for _IRR_COMPILE_WITH_OGLES1_                                        
-- Looking for _IRR_COMPILE_WITH_OGLES1_ - not found                            
-- Looking for _IRR_COMPILE_WITH_OGLES2_                                        
-- Looking for _IRR_COMPILE_WITH_OGLES2_ - not found                            
-- Looking for _IRR_COMPILE_WITH_OPENGL_                                        
-- Looking for _IRR_COMPILE_WITH_OPENGL_ - found                                
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libGL.so                             
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so                                                         
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found                                                 
-- Looking for gethostbyname            
-- Looking for gethostbyname - found    
-- Looking for connect                  
-- Looking for connect - found          
-- Looking for remove                   
-- Looking for remove - found           
-- Looking for shmat                    
-- Looking for shmat - found            
-- Found X11: /usr/lib/x86_64-linux-gnu/libX11.so                               
CMake Error at CMakeLists.txt:32 (install):                                     
  install TARGETS given target "IrrlichtMt" which does not exist in this        
  directory.                            

```
$ cmake --version
cmake version 3.7.2

Trying to compile for old Debian Stretch

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,8 @@ project(Irrlicht
 
 message(STATUS "*** Building IrrlichtMt ${PROJECT_VERSION} ***")
 
+include(GNUInstallDirs)
+
 if(ANDROID)
 	set(sysname Android)
 elseif(APPLE)
@@ -34,13 +36,7 @@ if(BUILD_EXAMPLES)
 	add_subdirectory(examples)
 endif()
 
-# Installation of library and headers.
-include(GNUInstallDirs)
-install(TARGETS ${INSTALL_TARGETS}
-	EXPORT IrrlichtMt-export
-	DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-)
-
+# Installation of headers.
 install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/"
 	DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/irrlichtmt"
 )
@@ -53,7 +49,7 @@ install(EXPORT IrrlichtMt-export
 )
 
 include(CMakePackageConfigHelpers)
-configure_package_config_file("${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in"
+configure_package_config_file("${PROJECT_SOURCE_DIR}/Config.cmake.in"
 	"${CMAKE_CURRENT_BINARY_DIR}/IrrlichtMtConfig.cmake"
 	INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/IrrlichtMt"
 	NO_SET_AND_CHECK_MACRO

--- a/source/Irrlicht/CMakeLists.txt
+++ b/source/Irrlicht/CMakeLists.txt
@@ -408,8 +408,6 @@ else()
 	set(INSTALL_TARGETS IrrlichtMt)
 endif()
 
-# Installation of library and headers.
-include(GNUInstallDirs)
 install(TARGETS ${INSTALL_TARGETS}
 	EXPORT IrrlichtMt-export
 	DESTINATION "${CMAKE_INSTALL_LIBDIR}"

--- a/source/Irrlicht/CMakeLists.txt
+++ b/source/Irrlicht/CMakeLists.txt
@@ -402,6 +402,7 @@ if(WIN32)
 	set_target_properties(IrrlichtMt PROPERTIES PREFIX "") # for DLL name
 endif()
 
+# Installation of library
 if(ANDROID)
 	set(INSTALL_TARGETS IrrlichtMt native_app_glue)
 else()

--- a/source/Irrlicht/CMakeLists.txt
+++ b/source/Irrlicht/CMakeLists.txt
@@ -403,7 +403,14 @@ if(WIN32)
 endif()
 
 if(ANDROID)
-	set(INSTALL_TARGETS IrrlichtMt native_app_glue PARENT_SCOPE)
+	set(INSTALL_TARGETS IrrlichtMt native_app_glue)
 else()
-	set(INSTALL_TARGETS IrrlichtMt PARENT_SCOPE)
+	set(INSTALL_TARGETS IrrlichtMt)
 endif()
+
+# Installation of library and headers.
+include(GNUInstallDirs)
+install(TARGETS ${INSTALL_TARGETS}
+	EXPORT IrrlichtMt-export
+	DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+)

